client/components/platform-checkout/hooks/use-phone-number-validity.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from 'react';
+
+// hook for checking the validity of phone number.
+const usePhoneNumberValidity = ( element, phoneNumber ) => {
+	const [ isValidNumber, setIsValidNumber ] = useState( false );
+
+	useEffect( () => {
+		const chechValidity = () => {
+			const instance = window.intlTelInputGlobals.getInstance( element );
+			setIsValidNumber( instance.isValidNumber() );
+		};
+
+		if ( element ) {
+			chechValidity();
+			element.addEventListener( 'countrychange', chechValidity );
+		}
+
+		return () => {
+			if ( element ) {
+				element.removeEventListener( 'countrychange', chechValidity );
+			}
+		};
+	}, [ element, phoneNumber ] );
+
+	return {
+		isValidNumber,
+	};
+};
+
+export default usePhoneNumberValidity;
```

Comment:
This is not used anywhere yet. This needs to be hooked to the `Place Order` button. When that button is clicked other form fields are validated. 
This will be handled with the issue when we will combine this UI with the backend API to create the user in `woopay` platform. Created this hook in advance to save some work later :smile: